### PR TITLE
Introduce docker image

### PR DESCRIPTION
**Pull request by vasiliy-zaznobin**
Introduce Dockerfile to create a docker image
Introduce .buildkite/* to create cybercoredev/uniswap-v2-core images
Introduce .buildkite/steps/deploy-test.sh to check all uniswap-v2-core tests with cybercoredev/proxy:latest by uping its proxy/docker-compose-test.yml

### CODE REVIEW

File: deploy-test.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
Comment (s-medvedev) [inline deploy-test.sh:1]: I suggest add flags `set -euo pipefail` at the begin of file.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM ubuntu:20.04
+
+RUN apt update && \
+    DEBIAN_FRONTEND=noninteractive apt -y install \
+            software-properties-common npm && \
+    npm install -g yarn && \
+    rm -rf /var/lib/apt/lists/* && \
+    yarn
```
Comment (s-medvedev) [inline Dockerfile]: It seems te me that yarn should be called after copy. Do it with yarn compile.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+RUN apt update && \
+    DEBIAN_FRONTEND=noninteractive apt -y install \
+            software-properties-common git npm && \
+    npm install -g yarn waffle && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY . /opt
+WORKDIR /opt
+RUN yarn
+RUN yarn compile
```
Comment (s-medvedev) [inline Dockerfile]: Change to `RUN yarn && yarn compile`

File: .buildkite/steps/deploy-test.sh [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+function cleanup_docker {
+    echo "Cleanup docker-compose..."
+    docker-compose -f node-and-proxy.yml down
+    echo "Cleanup docker-compose done."
+}
+trap cleanup_docker EXIT
+sleep 10
+
+REVISION=$(git rev-parse HEAD)
```
Comment (s-medvedev) [inline .buildkite/steps/deploy-test.sh]: Use `export ...` for this variable.

File: .buildkite/steps/deploy-test.sh [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    echo "Cleanup docker-compose done."
+}
+trap cleanup_docker EXIT
+sleep 10
+
+export PROXY_URL=http://127.0.0.1:9090/solana
+
+echo "Wait proxy..." && wait-for-proxy "$PROXY_URL"
+
+./deploy-test.sh
```
Comment (s-medvedev) [inline .buildkite/steps/deploy-test.sh]: You should run deploy-test.sh from container (like in solana-program-library/./buildkite/steps/deploy-tesst.sh).

**Comment by vasiliy-zaznobin** [comment]
Most of the tests passed, the rest fell off by timeout

**Pull request merged**